pkg/operator/ceph/object/objectstore.go

```diff
@@ -88,18 +98,89 @@ func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
 	return nil
 }
 
+func checkZoneIsMaster(context *Context, realmArg string, zoneGroupArg string, zoneArg string) (bool, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	masterZoneID, err := decodeMasterZoneID(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse master zone id")
+	}
+
+	zoneOutput, err := RunAdminCommandNoRealm(context, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone")
+	}
+	zoneID, err := decodeID(zoneOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse zone id")
+	}
+
+	if zoneID == masterZoneID {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func getZoneEndpoints(context *Context, realmArg string, zoneGroupArg string, zoneName string, serviceEndpoint string) (string, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zones, err := decodeZonesList(zoneGroupOutput)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse zones list")
+	}
+
+	numZones := len(zones)
+	zoneEndpoints := ""
+	for i := 0; i < numZones; i++ {
```

Comment:
nit: The more idiomatic way in golang to iterate would be with the `range` and not even use the index. For example:
```
for _, zone := range zones {
  numEndpoints := len(zone.Endpoints)
  for _, endpoint := range zone.Endpoints {
    if endpoint != serviceEndpoint {
      zoneEndpoints += endpoint + ","
    }
  }
}
```